### DISCUSSION TITLE
Cfclient crashing related to Qt

### DISCUSSION THREAD

**Discussion by SkogsTomte** [bitcraze/discussions#898] [Q&A]
I get the following message while trying to run cfclient:

Traceback (most recent call last):
  File "/usr/local/bin/cfclient", line 33, in <module>
    sys.exit(load_entry_point('cfclient', 'console_scripts', 'cfclient')())
  File "/root/crazyflie-clients-python/src/cfclient/gui.py", line 192, in main
    main_window = MainUI()
  File "/root/crazyflie-clients-python/src/cfclient/ui/main.py", line 270, in __init__
    self.loaded_tab_toolboxes = self.create_tab_toolboxes(self.tabs_menu_item,
  File "/root/crazyflie-clients-python/src/cfclient/ui/main.py", line 327, in create_tab_toolboxes
    tab_toolbox = tab_class(cfclient.ui.pluginhelper)
  File "/root/crazyflie-clients-python/src/cfclient/ui/tabs/locopositioning_tab.py", line 465, in __init__
    self._set_up_plots()
  File "/root/crazyflie-clients-python/src/cfclient/ui/tabs/locopositioning_tab.py", line 493, in _set_up_plots
    self._plot_layout.addWidget(self._plot_3d.native)
TypeError: addWidget(self, QWidget, stretch: int = 0, alignment: Union[Qt.Alignment, Qt.AlignmentFlag] = Qt.Alignment()): argument 1 has unexpected type 'CanvasBackend'

I installed it using these instructions: https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/installation/install/

both using pip and building from source, but I still get the same error.

**Comment by krichardsson** [comment]
Which PyQT version do you have?
run `pip list` and copy/paste anything named PyQt*** here

**Comment by SkogsTomte** [comment]
Here is the output. I should also mention that i am on Ubuntu 22.04

PyQt5                              5.15.6
PyQt5-sip                        12.9.1
pyqtgraph                        0.13.3
qt-dotgraph                     2.2.2
qt-gui                               2.2.2
qt-gui-cpp                        2.2.2
qt-gui-py-common           2.2.2
qtm                                  2.1.2

**Comment by krichardsson** [comment]
That looks reasonable to me, not sure what is going on. 

The error originates from trying to use vispy with qt5. I have
```
PyQt5              5.15.9
PyQt5-Qt5          5.15.2
PyQt5-sip          12.12.2
vispy              0.13.0
```
and that works. Check that your vispy is the right version, but I expect it is.

Qt5 is pretty old and there are some random problems reported that are hard to understand. We are moving to Qt6 and there is a branch you can try if you feel brave. The name of the branch is "krichardsson/qt6-v2", just check it out and run `pip install -e .`. Note that I have only done some basic testing and there might be some bugs, but most of the stuff seems to be OK.

**Comment by SkogsTomte** [comment]
Yes, I have the same vispy version. I tried the Qt6 branch but then it complains about xcb instead, even though it is installed.

qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
Available platform plugins are: vkkhrdisplay, minimalegl, vnc, minimal, offscreen, linuxfb, wayland-egl, eglfs, wayland, xcb.

**Comment by krichardsson** [comment]
Unfortunately I'm not really sure what to recommend. Ubuntu 22.04 should be fine and the versions you have seems to be OK. 
I assume you are a some newish python version as well? 
Maybe try some other python version and reinstall stuff?
Sorry I can not help more.

**Comment by SkogsTomte** [comment]
Yes, I am using python 3.10.12 It works well on another system I have which is almost identical, so I can't really wrap my head around this.
Anyway, thank you for trying to help me!
